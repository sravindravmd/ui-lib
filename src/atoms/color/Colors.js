import React from 'react'
import { StyledColorsContainer } from './Color.styles'
import { Color } from './Color'
import PropTypes from 'prop-types'

export const Colors = ({ colors = [] }) => {
  return (
    <StyledColorsContainer>
      {colors.map(({ name, path, color }) => (
        <Color color={color} name={name} path={path} key={name} />
      ))}
    </StyledColorsContainer>
  )
}

Colors.propTypes = {
  colors: PropTypes.array,
}
