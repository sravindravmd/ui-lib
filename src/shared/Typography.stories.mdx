import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks'
import { StyledH1, StyledH2 } from '../molecules/typography/Typography.styles'

<Meta title='Shared/Theme/Typography' component={StyledH1}/>

# Typography

```import { theme } from '@teamfabric/copilot-ui'```



