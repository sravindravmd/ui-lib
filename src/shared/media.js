import { css } from 'styled-components'
import theme from './theme.js'

export default {
  all: (...args) => css(...args),
  small: (...args) => css`
    @media (max-width: ${theme.breakpoints.values.md}px) {
      ${css(...args)};
    }
  `,
  gtSmall: (...args) => css`
    @media (min-width: ${theme.breakpoints.values.md + 1}px) {
      ${css(...args)};
    }
  `,
  medium: (...args) => css`
    @media (min-width: ${theme.breakpoints.values.md +
      1}px and max-width ${theme.breakpoints.values.lg}px) {
      ${css(...args)}
    }
  `,
  gtMedium: (...args) => css`
    @media (min-width: ${theme.breakpoints.values.md + 1}px) {
      ${css(...args)};
    }
  `,
  large: (...args) => css`
    @media (min-width: ${theme.breakpoints.values.lg + 1}px) {
      ${css(...args)};
    }
  `,
  gtLarge: (...args) => css`
    @media (min-width: ${theme.breakpoints.values.lg + 1}px) {
      ${css(...args)};
    }
  `,
}
