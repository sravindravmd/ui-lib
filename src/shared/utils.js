export const lineBreak = string => string.replace(/\n/gi, '\n')
