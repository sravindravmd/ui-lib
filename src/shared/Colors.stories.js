import React from 'react'
import { Colors } from '../atoms/color/Colors'
import { theme } from './index'
export default {
  title: 'Shared/Theme/Color',
  component: Colors,
}

const brandPrimary = [
  {
    color: theme.palette.brand.primary.charcoal,
    name: 'Primary Default',
    path: 'theme.palette.brand.primary.charcoal',
  },
  {
    color: theme.palette.brand.primary.gray,
    name: 'Primary Secondary',
    path: 'theme.palette.brand.primary.gray',
  },
  {
    color: theme.palette.brand.primary.white,
    path: 'theme.palette.brand.primary.white',
  },
]
const brandSecondaryColors = [
  {
    color: theme.palette.brand.secondary.purple,
    name: 'Purple',
    path: 'theme.palette.brand.secondary.purple',
  },
  {
    color: theme.palette.brand.secondary.pink,
    name: 'Pink',
    path: 'theme.palette.brand.secondary.pink',
  },
  {
    color: theme.palette.brand.secondary.yellow,
    name: 'Yellow',
    path: 'theme.palette.brand.secondary.yellow',
  },
  {
    color: theme.palette.brand.secondary.green,
    name: 'Green',
    path: 'theme.palette.brand.secondary.green',
  },
  {
    color: theme.palette.brand.secondary.blue,
    name: 'Blue',
    path: 'theme.palette.brand.secondary.blue',
  },
]
const uiCtaStatesColors = [
  {
    color: theme.palette.ui.states.active,
    name: 'Active',
    path: 'theme.palette.ui.states.active',
  },
  {
    color: theme.palette.ui.states.hover,
    name: 'Hover',
    path: 'theme.palette.ui.states.hover',
  },
  {
    color: theme.palette.ui.states.disabled,
    name: 'Disabled',
    path: 'theme.palette.ui.states.disabled',
  },
  {
    color: theme.palette.ui.states.pressed,
    name: 'Pressed',
    path: 'theme.palette.ui.states.pressed',
  },
  {
    color: theme.palette.ui.cta.yellow,
    name: 'Warning',
    path: 'theme.palette.ui.cta.yellow',
  },
  {
    color: theme.palette.ui.cta.green,
    name: 'Success',
    path: 'theme.palette.ui.cta.green',
  },
  {
    color: theme.palette.ui.cta.red,
    name: 'Error',
    path: 'theme.palette.ui.cta.red',
  },
  {
    color: theme.palette.ui.cta.blue,
    name: 'Blue',
    path: 'theme.palette.ui.cta.blue',
  },
]
const uiNeutral = [
  {
    color: theme.palette.ui.neutral.grey1,
    name: 'Grey1',
    path: 'theme.palette.ui.neutral.grey1',
  },
  {
    color: theme.palette.ui.neutral.grey2,
    name: 'Grey2',
    path: 'theme.palette.ui.neutral.grey2',
  },
  {
    color: theme.palette.ui.neutral.grey3,
    name: 'Grey3',
    path: 'theme.palette.ui.neutral.grey3',
  },
  {
    color: theme.palette.ui.neutral.grey4,
    name: 'Grey4',
    path: 'theme.palette.ui.neutral.grey4',
  },
  {
    color: theme.palette.ui.neutral.grey5,
    name: 'Grey5',
    path: 'theme.palette.ui.neutral.grey5',
  },
  {
    color: theme.palette.ui.neutral.grey6,
    name: 'Grey6',
    path: 'theme.palette.ui.neutral.grey6',
  },
  {
    color: theme.palette.ui.neutral.grey7,
    name: 'Grey7',
    path: 'theme.palette.ui.neutral.grey7',
  },
  {
    color: theme.palette.ui.neutral.grey8,
    name: 'Grey8',
    path: 'theme.palette.ui.neutral.grey8',
  },
]

const Template = args => <Colors {...args} />

export const BrandPrimaryColor = Template.bind({})
BrandPrimaryColor.args = {
  colors: brandPrimary,
}
export const BrandSecondaryColor = Template.bind({})
BrandSecondaryColor.args = {
  colors: brandSecondaryColors,
}
export const UiCtaStatesColor = Template.bind({})
UiCtaStatesColor.args = {
  colors: uiCtaStatesColors,
}
export const UiNeutralColor = Template.bind({})
UiNeutralColor.args = {
  colors: uiNeutral,
}
