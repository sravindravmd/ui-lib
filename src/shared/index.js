import theme from './theme.js'
import media from './media'

const FONT_NAMES = [
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6',
  'subtitle1',
  'subtitle2',
  'subtitle3',
  'body',
  'kicker',
  'caption',
  'link',
]

FONT_NAMES.forEach(name => {
  const css = []
  Object.keys(theme.typography[name]).forEach(styleKey => {
    const dashKey = styleKey.replace(/.[A-Z]/, match => {
      return match[0] + '-' + match[1].toLowerCase()
    })
    css.push(`${dashKey}: ${theme.typography[name][styleKey]}`)
  })
  theme.typography[name].css = css.join(';\n')
})

export { theme, media }
