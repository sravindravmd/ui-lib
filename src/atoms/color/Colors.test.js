import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Colors } from './Colors'
import { theme } from '../../shared'

afterEach(cleanup)

const colors = [
  {
    color: theme.palette.ui.neutral.grey1,
    name: 'Grey1',
    path: 'theme.palette.ui.neutral.grey1',
  },
  {
    color: theme.palette.ui.neutral.grey2,
    name: 'Grey2',
    path: 'theme.palette.ui.neutral.grey2',
  },
]

describe('<Colors/>', () => {
  it('renders Colors component correctly', () => {
    const { colorSnap } = render(<Colors colors={colors} />)
    expect(colorSnap).toMatchSnapshot()
  })
})
