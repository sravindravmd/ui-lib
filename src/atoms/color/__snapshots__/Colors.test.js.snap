// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Colors/> renders Colors component correctly 1`] = `undefined`;
